storage/src/backend/connection.rs

```diff
@@ -363,7 +348,55 @@ impl Connection {
                 });
             }
         }
-        // TODO: check mirrors' health
+        // Check mirrors' health
+        for mirror in connection.mirrors.iter() {
+            let conn = connection.clone();
+            let connect_timeout = config.connect_timeout;
+            // Spawn thread to update the health status of mirror server
+            let mirror_cloned = mirror.clone();
+            thread::spawn(move || {
+                let mirror_health_url = if mirror_cloned.config.ping_url.is_empty() {
+                    format!("{}/v2", mirror_cloned.config.host)
+                } else {
+                    mirror_cloned.config.ping_url.clone()
+                };
+                info!("Mirror health checking url: {}", mirror_health_url);
+
+                loop {
+                    // Try to recover the mirror server when it is unavailable.
+                    if !mirror_cloned.status.load(Ordering::Relaxed) {
+                        info!(
+                            "Mirror {} unhealthy, try to recover",
+                            mirror_cloned.config.host
+                        );
+                        let client = Client::new();
+                        let _ = client
```

Comment:
Why ignore the error? maybe we will encounter a network error.

Reply:
Because the mirror server is not available when we get an error.  Maybe add a `warn!` log line for it?